### PR TITLE
Feat: Register drivers and operators

### PR DESCRIPTION
**Pull request by dmutoni**
#### What does this PR do?
* Adds register drivers and operators
#### Description of Task to be completed?
* The administrator should be able to register a new driver or a new operator.
* An email containing a random password should be sent to the provided email.
* Appropriate error messages are displayed on unsuccessful attempts
#### How should this be manually tested?
* Clone the repo
* Run npm install to install modules
* Run npm run dev to start the development server and the database
* Go to the endpoint `"/api/users/register"`
#### Any background context you want to provide?
image

#### What are the relevant pivotal tracker/Trello stories?
* [Trello](https://trello.com/c/pB9AVElH/10-register-drivers-operators)

### CODE REVIEW

**Review by MikeManzi — COMMENTED** [review]
Registration works. However, I'm getting this error when trying to send the email.
image

**Review by MikeManzi — APPROVED** [review]
LGTM!

**Review by kelliaUmuhire — APPROVED** [review]

File: src/controllers/userController.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            subject: res.__("login_credentials_message"),
+            text: `${res.__("your_login_credentials_message")}.\nEmail: ${email}\nPassword: ${password}`,
+        }
+        transporter.sendMail(mailOptions, (error, info) => {
+            if (error) {
+                console.log(error)
+            } else {
+                console.log(`Email sent: ${info.response}`)
+            }
+        })
```
Comment (freddykaberuka) [inline src/controllers/userController.ts]: create this email sender as a function.
example:
replace the highlighted lines of code by calling sendEmail function in createUser func
`await sendEmail(email, password, res)`
and create a send email function
````
const sendEmail = async (email: string, password: string, res: Response) => {
  const transporter = nodemailer.createTransport({
    service: "gmail",
    auth: {
      user: process.env.EMAIL_ADDRESS,
      pass: process.env.EMAIL_PASSWORD,
    },
  })

  const mailOptions = {
    from: process.env.EMAIL_ADDRESS,
    to: email,
    subject: res.__("login_credentials_message"),
    text: `${res.__("your_login_credentials_message")}.\nEmail: ${email}\nPassword: ${password}`,
  }

  transporter.sendMail(mailOptions, (error, info) => {
    if (error) {
      console.log(error)
    } else {
      console.log(`Email sent: ${info.response}`)
    }
  })
}
```

File: src/controllers/userController.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        const user: UserModel | null = await UserModel.findOne({
+            where: {
+                email: email,
+            },
+        })
+        return user
+    } catch (error) {
+        return null
+    }
+}
```
Comment (freddykaberuka) [inline src/controllers/userController.ts]: function naming is `findUserByEmail` not `findUserById`

**Review by freddykaberuka — CHANGES_REQUESTED** [review]
Hi @dmutoni,
Good job so far! 👏👏👏, however, there are some changes needed to make your codes DRY.
Please check requested comment

File: src/controllers/userController.ts (hunk 2 above)
Comment (dmutoni) [reply src/controllers/userController.ts]: you are right, thank you

**Pull request merged**